Tests/Marketplace/search_and_install_packs.py

```diff
@@ -413,381 +600,348 @@ def install_packs(client: demisto_client,
-    except Exception:
-        logging.exception('The request to install packs has failed.')
-        sys.exit(1)
+    api_data = get_pack_dependencies(client=client, pack_id=pack_id)
 
+    if not api_data:
+        return pack_id, []
 
-def search_and_install_packs_and_their_dependencies_private(test_pack_path: str,
-                                                            pack_ids: list,
-                                                            client: demisto_client):
-    """ Searches for the packs from the specified list, searches their dependencies, and then installs them.
-    Args:
-        test_pack_path (str): Path of where the test packs are located.
-        pack_ids (list): A list of the pack ids to search and install.
-        client (demisto_client): The client to connect to.
-
-    Returns (list, bool):
-        A list of the installed packs' ids, or an empty list if is_nightly == True.
-        A flag that indicates if the operation succeeded or not.
-    """
-    host = client.api_client.configuration.host
-
-    logging.info(f'Starting to search and install packs in server: {host}')
-
-    install_packs_private(client, host, pack_ids, test_pack_path)
+    dependencies_data: list[dict] = create_dependencies_data_structure(
+        response_data=api_data.get("dependencies", []),
+        dependants_ids=[pack_id],
+        checked_packs=[pack_id],
+    )
+    pack_api_data = api_data["packs"][0]
+    current_packs_to_install = [pack_api_data]
 
-    return SUCCESS_FLAG
+    if dependencies_data:
+        logging.debug(
+            f"Found dependencies for '{pack_id}': {[dependency['id'] for dependency in dependencies_data]}"
+        )
 
+        for dependency in dependencies_data:
+            if is_pack_deprecated(
+                pack_id=dependency["id"],
+                production_bucket=production_bucket,
+                pack_api_data=dependency,
+            ):
+                logging.critical(
+                    f"Pack '{pack_id}' depends on pack '{dependency['id']}' which is a deprecated pack."
+                )
+                return pack_id, None
+            else:
+                current_packs_to_install.append(dependency)
 
-def search_and_install_packs_and_their_dependencies(pack_ids: list,
-                                                    client: demisto_client, hostname: str | None = None,
-                                                    multithreading: bool = True,
-                                                    production_bucket: bool = True):
+    dependencies = [
+        get_pack_installation_request_data(
+            pack_id=pack["id"],
+            pack_version=pack["extras"]["pack"]["currentVersion"],
+        )
+        for pack in current_packs_to_install
+    ]
+    return pack_id, dependencies
+
+
+def flatten_dependencies(pack_id: str, pack_dependencies: list[dict], all_packs_dependencies: dict[str, list[dict]]):
```

Comment:
Please add a return value type hint.
It's important here, as you call it recursively, and extending a list.